tests/subsys/fs/nvs/src/main.c

```diff
@@ -245,6 +246,138 @@ void test_nvs_gc(void)
 	zassert_true(err == 0,  "nvs_clear call failure: %d", err);
 }
 
+static void memset_16(u16_t *buf, u16_t data, size_t item_count)
+{
+	for (size_t i = 0; i < item_count; i++) {
+		buf[i] = data;
+	}
+}
+
+static void write_content_16(u16_t max_id, u16_t begin, u16_t end,
+			     struct nvs_fs *fs)
+{
+	u16_t buf[16];
+	ssize_t len;
+
+	for (u16_t i = begin; i < end; i++) {
+		u8_t id = (i % max_id);
+		u16_t id_data = id + max_id * (i / max_id);
+
+		memset_16(buf, id_data, ARRAY_SIZE(buf));
```

Comment:
Why not use libc's memset? Is there something about the 16 bit width that matters here that means we cannot just use an 8 bit `id_data`? It doesn't look like the answer is yes, since `max_id` is just 10 below.

Reply:
reworked